src/codegen/generators/expression_generator.rs

```diff
@@ -2266,6 +2266,19 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
             ))
         }?;
 
+        // for arrays of struct we cannot flatten the expression list
+        // to generate the passed structs we need an expression list of assignments
+        // flatten_expression_list will will return a vec of only assignments
+        let elements = if let DataTypeInformation::Struct { .. } = inner_type.get_type_information()
```

Comment:
you could use `if inner_type.get_type_information().is_struct()` which is a bit less code